resources/bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/localstore/FileSystemResourceManager.java

```diff
@@ -1243,7 +1243,7 @@ public void write(IFile target, InputStream content, IFileInfo fileInfo, int upd
 	}
 
 	private void prepareWrite(IFile target, IFileInfo fileInfo, int updateFlags, boolean append,
-			Resource targetResource, IFileStore store) throws CoreException {
+			Resource targetResource, IFileStore store, boolean optimistic) throws CoreException {
```

Comment:
The name of the parameter should be more precise. For example `assumeParentDirectoryExists`.
Then you don't have to add a comment below about what the method is optimistic about :)

Reply:
done.